scripts/illumina_run_parameter_parser.py

```diff
@@ -0,0 +1,230 @@
+                except:
+                    sys.stderr.write("No RunInfo.xml found")
+                    sys.exit(2)
+        elif outart.type == 'ResultFile' and outart.name == 'Run Parameters':
+            try:
+                lims.upload_new_file(outart, "{}/RunParameters.xml".format(run_dir))
+            except:
+                try:
+                    lims.upload_new_file(outart, "{}/runParameters.xml".format(run_dir))
+                except:
+                    sys.stderr.write("No RunParameters.xml found")
+                    sys.exit(2)
+
+
+def lims_for_nextseq(process, run_dir):
+    # Parse run
+    runParserObj, RunParametersParserObj = parse_run(run_dir)
+    # Attach RunInfo.xml and RunParamters.xml
+    attach_xml(process, run_dir)
+    # Set values for LIMS UDFs
+    process.udf['Finish Date'] = datetime.strptime(RunParametersParserObj.data['RunParameters']['RunEndTime'][:10],"%Y-%m-%d").date() if 'RunEndTime' in RunParametersParserObj.data['RunParameters'].keys() and RunParametersParserObj.data['RunParameters']['RunEndTime'] != '' else datetime.now().date()
+    process.udf['Run Type'] = "NextSeq 2000 {}".format(RunParametersParserObj.data['RunParameters']['FlowCellMode'].split(' ')[2])
+    process.udf['Chemistry'] = "NextSeq 2000 {}".format(RunParametersParserObj.data['RunParameters']['FlowCellMode'].split(' ')[2])
+    planned_cycles = sum(list(map(int, RunParametersParserObj.data['RunParameters']['PlannedCycles'].values())))
+    completed_cycles = sum(list(map(int, RunParametersParserObj.data['RunParameters']['CompletedCycles'].values())))
+    process.udf['Status'] = "Cycle {} of {}".format(completed_cycles, planned_cycles)
+    process.udf['Flow Cell ID'] = RunParametersParserObj.data['RunParameters']['FlowCellSerialNumber']
+    process.udf['Experiment Name'] = RunParametersParserObj.data['RunParameters']['FlowCellSerialNumber']
+    process.udf['Read 1 Cycles'] = int(RunParametersParserObj.data['RunParameters']['PlannedCycles']['Read1'])
+    process.udf['Index 1 Read Cycles'] = int(RunParametersParserObj.data['RunParameters']['PlannedCycles']['Index1'])
+    process.udf['Index 2 Read Cycles'] = int(RunParametersParserObj.data['RunParameters']['PlannedCycles']['Index2'])
+    process.udf['Read 2 Cycles'] = int(RunParametersParserObj.data['RunParameters']['PlannedCycles']['Read2'])
+    process.udf['Run ID'] = runParserObj.runinfo.data['Id']
+    process.udf['Reagent Cartridge ID'] = RunParametersParserObj.data['RunParameters']['CartridgeSerialNumber']
+    # Put in LIMS
+    process.put()
+
+
+def lims_for_miseq(process, run_dir):
+    # Parse run
+    runParserObj, RunParametersParserObj = parse_run(run_dir)
+    # Set values for LIMS UDFs
+    process.udf['Finish Date'] = datetime.now().date()
+    if RunParametersParserObj.data['RunParameters']['Setup']['SupportMultipleSurfacesInUI'] == 'true' and unParametersParserObj.data['RunParameters']['Setup']['NumTilesPerSwath'] == '19':
```

Comment:
In the following if else there are some `unParametersParserObj`, I guess it should be `RunParametersParserObj`?

Reply:
Thanks! I must be tired...

Reply:
😄